src/systems/filecoin_vm/interpreter/vm_interpreter.go

```diff
@@ -177,7 +178,9 @@ func (vmi *VMInterpreter_I) ApplyMessage(
 
 	// Check sender balance.
 	gasLimitCost := _gasToFIL(message.GasLimit(), message.GasPrice())
-	totalCost := message.Value() + actor.TokenAmount(gasLimitCost)
+	networkTxnFee := indices.Indices_FromStateTree(inTree).NetworkTransactionFee(
+		inTree.GetActorCodeID_Assert(message.To()), message.Method())
+	totalCost := message.Value() + gasLimitCost + networkTxnFee
```

Comment:
nice 👍

Reply:
I don't think this is good. Even if there was some additional fee as part of sending a message, it still has to fit within the gas limit so a sender can bound their cost.

Reply:
> it still has to fit within the gas limit so a sender can bound their cost

I don't see how that follows. I think the sender can bound their cost even if it doesn't fit within the gas limit, the bound just has to be increased to include the maximum transaction fee for the given method.